python/ray/data/preprocessors/chain.py

```diff
@@ -1,15 +1,21 @@
+import warnings
 from typing import TYPE_CHECKING, Union
 
 from ray.air.util.data_batch_conversion import BatchFormat
 from ray.data import Dataset, DatasetPipeline
 from ray.data.preprocessor import Preprocessor
-from ray.util.annotations import PublicAPI
+from ray.util.annotations import Deprecated
 
 if TYPE_CHECKING:
     from ray.air.data_batch_type import DataBatchType
 
+CHAIN_DEPRECATION_MESSAGE = (
+    "The Chain preprocessor is deprecated as of Ray 2.7. Instead, manually apply your "
```

Comment:
Same as above

Reply:
This one is just a warning since this is a new deprecation.